haystack/components/preprocessors/recursive_splitter.py

```diff
@@ -0,0 +1,296 @@
+            text will be split into sentences using a custom sentence tokenizer based on NLTK.
+            See: haystack.components.preprocessors.sentence_tokenizer.SentenceSplitter.
+            If no separators are provided, the default separators ["\n\n", "sentence", "\n", " "] are used.
+
+        :raises ValueError: If the overlap is greater than or equal to the chunk size or if the overlap is negative, or
+                            if any separator is not a string.
+        """
+        self.split_length = split_length
+        self.split_overlap = split_overlap
+        self.split_units = split_units
+        self.separators = separators if separators else ["\n\n", "sentence", "\n", " "]  # default separators
+        self.sentence_tokenizer_params = sentence_splitter_params
+        self._check_params()
+        if "sentence" in self.separators:
+            sentence_splitter_params = sentence_splitter_params or {"keep_white_spaces": True}
+            self.nltk_tokenizer = self._get_custom_sentence_tokenizer(sentence_splitter_params)
+
+    def _check_params(self):
+        if self.split_length < 1:
+            raise ValueError("Split length must be at least 1 character.")
+        if self.split_overlap < 0:
+            raise ValueError("Overlap must be greater than zero.")
+        if self.split_overlap >= self.split_length:
+            raise ValueError("Overlap cannot be greater than or equal to the chunk size.")
+        if not all(isinstance(separator, str) for separator in self.separators):
+            raise ValueError("All separators must be strings.")
+
+    @staticmethod
+    def _get_custom_sentence_tokenizer(sentence_splitter_params: Dict[str, Any]):
+        from haystack.components.preprocessors.sentence_tokenizer import SentenceSplitter
+
+        return SentenceSplitter(**sentence_splitter_params)
+
+    def _apply_overlap(self, chunks: List[str]) -> List[str]:
+        """
+        Applies an overlap between consecutive chunks if the chunk_overlap attribute is greater than zero.
+
+        :param chunks: List of text chunks.
+        :returns:
+            The list of chunks with overlap applied.
+        """
+        overlapped_chunks = []
+
+        for idx, chunk in enumerate(chunks):
+            if idx == 0:
+                overlapped_chunks.append(chunk)
+                continue
+            overlap_start = max(0, self._chunk_length(chunks[idx - 1]) - self.split_overlap)
+            if self.split_units == "word":
+                word_chunks = chunks[idx - 1].split()
+                overlap = " ".join(word_chunks[overlap_start:])
+            else:
+                overlap = chunks[idx - 1][overlap_start:]
+            if overlap == chunks[idx - 1]:
+                logger.warning(
+                    "Overlap is the same as the previous chunk. "
+                    "Consider increasing the `split_overlap` parameter or decreasing the `split_length` parameter."
+                )
```

Comment:
Is there a test for this? I couldn't find it when quickly looking through the tests?

Reply:
Also I think the suggestion should swap the two parameter names right?
```suggestion
            if overlap == chunks[idx - 1]:
                logger.warning(
                    "Overlap is the same as the previous chunk. "
                    "Consider increasing the `split_length` parameter or decreasing the `split_overlap` parameter."
                )
```

Reply:
There is, it's called `test_apply_overlap_with_overlap_capturing_completely_previous_chunk` - just need to update the warning string